### PR TITLE
validator transaction API

### PR DESCRIPTION
**Pull request by zjma**
### Description

A naive validator transaction API that exposes general transaction info and events.

### TODOs
- Upgrade proto version. cc @larry-aptos we need to coordinate with @hariria as well

### CODE REVIEW

**Comment by zjma** [comment]
fyi @bowenyang007 this feature renamed to "validator transaction".
I have applied the renaming in this PR (including running `./scripts/build_protos.sh`), but please still do a second pass 🙏

File: protos/proto/aptos/transaction/v1/transaction.proto [hunk 1]
```diff
@@ -50,6 +50,7 @@ message Transaction {
     TRANSACTION_TYPE_BLOCK_METADATA = 2;
     TRANSACTION_TYPE_STATE_CHECKPOINT = 3;
     TRANSACTION_TYPE_USER = 4;
+    TRANSACTION_TYPE_VALIDATOR = 20;
```
Comment (grao1991) [inline protos/proto/aptos/transaction/v1/transaction.proto:53]: should this be 5?

File: protos/proto/aptos/transaction/v1/transaction.proto [hunk 2]
```diff
@@ -59,6 +60,7 @@ message Transaction {
     GenesisTransaction genesis = 8;
     StateCheckpointTransaction state_checkpoint = 9;
     UserTransaction user = 10;
+    ValidatorTransaction validator = 21;
```
Comment (grao1991) [inline protos/proto/aptos/transaction/v1/transaction.proto:63]: should this be 11?

Comment (zjma) [reply protos/proto/aptos/transaction/v1/transaction.proto:63]: @bowenyang007

File: ecosystem/indexer-grpc/indexer-grpc-fullnode/src/stream_coordinator.rs [hunk 3]
```diff
@@ -313,6 +313,7 @@ impl IndexerStreamCoordinator {
                             sct.info.block_height = Some(block_height_bcs);
                             sct.info.epoch = Some(epoch_bcs);
                         },
+                        APITransaction::ValidatorTransaction(_) => todo!(),
```
[the remark this answers is not in the document]
Comment (zjma) [reply ecosystem/indexer-grpc/indexer-grpc-fullnode/src/stream_coordinator.rs:316]: we probably have to...

**Review by bowenyang007 — CHANGES_REQUESTED** [review]
looks good. please test in fullnode API before landing this.

**Review by bowenyang007 — APPROVED** [review]

File: protos/proto/aptos/transaction/v1/transaction.proto (hunk 2 above)
Comment (bowenyang007) [reply protos/proto/aptos/transaction/v1/transaction.proto:63]: doesn't matter. leaving some space might be good 🤷

Comment (bowenyang007) [reply protos/proto/aptos/transaction/v1/transaction.proto:63]: 11 works too.

File: protos/proto/aptos/transaction/v1/transaction.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (bowenyang007) [reply protos/proto/aptos/transaction/v1/transaction.proto:53]: i don't remember lol. 5 is fine.

**Review by zekun000 — APPROVED** [review]

**Pull request merged**